### ISSUE TITLE
read_until in 3.5 is backwards incompatible due to change of to argument name

### ISSUE THREAD

**Issue by carandraug** [pyserial/pyserial#634]
In version 3.5 of pySerial the argument names of `Serial.read_until` changed from:

    def read_until(self, terminator, size)

to 

    def read_until(self, expected, size)

This was done in commit daaf33edd1a054a52fa8ec297a5767d06f96a942. From the commit message I understand that the name was changed to better reflect the meaning. However, this change is backwards incompatible leading to:

    TypeError: read_until() got an unexpected keyword argument 'terminator'

**Comment by Abdel-dev-tech** [comment]
So it's better to downdate to latest stable release 3.4-4
the following resolve the probleme for me :
Unistall it from PyPI:

- pip uninstall pyserial or pip3 uninstall pyserial (Debian)

Then, install it from the package using :

- sudo apt-get install python-serial (python 2.7)

or 

- sudo apt-get install python3-serial (python 3)

**Comment by carandraug** [comment]
Yes, I know how to downgrade pyserial. And that solution works fine until there's a transitive dependency on pyserial>=3.5. Software that uses pyserial and named arguments can update themselves easily and simply not use named arguments to remain compatible with all pyserial releases. So all can be solved "easily" but one from outside still needs to figure out what was happening (pyserial changed argument names), dig through the history of pyserial to try and figure why it changed (and be surprised that the change of name really has no change of behaviour), then make a fix on their own packages and a new release (and reply for the next couple of years to their own users that do not use the very last release).

I guess this open issue was more of a "please don't rename argument names gratuitously as it breaks other people software. It's too late now but please think again next time".
